tempodb/backend/block_meta.go

```diff
@@ -337,3 +294,64 @@ func (dcs DedicatedColumns) Hash() uint64 {
 	}
 	return h.Sum64()
 }
+
+func (dcs DedicatedColumns) Size() int {
+	if len(dcs) == 0 {
+		return 0
+	}
+
+	b, _ := dcs.Marshal()
+	return len(b)
+}
+
+func (dcs DedicatedColumns) Marshal() ([]byte, error) {
+	if len(dcs) == 0 {
+		return nil, nil
+	}
+
+	// NOTE: This is a performance optimization to avoid re-unmarshalling the
+	// same data multiple times on the read.
+	return json.Marshal(dcs)
+}
+
+func (dcs DedicatedColumns) MarshalTo(data []byte) (n int, err error) {
+	bb, err := dcs.Marshal()
+	if err != nil {
+		return 0, err
+	}
+	copy(data, bb)
+
+	return len(bb), nil
+}
+
+func (dcs *DedicatedColumns) Unmarshal(data []byte) error {
+	if len(data) == 0 {
+		return nil
+	}
+
+	// NOTE: See UnmarshalJSON for a performance optimization to record the json
+	// bytes for reuse.
```

Comment:
Let's specifically call out "interning" the dedicated column config in this coment.